pyscriptjs/tests/integration/conftest.py

```diff
@@ -142,10 +142,18 @@ def run(self):
 @pytest.fixture(scope="session")
 def http_server(logger):
     class MyHTTPRequestHandler(SimpleHTTPRequestHandler):
+        def end_headers(self):
+            self.send_my_headers()
+            SimpleHTTPRequestHandler.end_headers(self)
+
+        def send_my_headers(self):
+            self.send_header("Cross-Origin-Embedder-Policy", "require-corp")
+            self.send_header("Cross-Origin-Opener-Policy", "same-origin")
+
```

Comment:
what happens if the web server doesn't set these headers?
I would like the following:
1. pyscript detects this case
2. it displays a nice error to the user with a link to the relevant docs
3. we have a test for that

I think it's fine to do it in a follow-up PRs though. Once this is merged, please open an issue so that we don't forget (and even better, fix it ;))

Reply:
well, but at the moment of writing we don't support the ServiceWorker approach yet, so we should show a very explicit warning/error.

> In this case, we can still run in a worker but can't read from stdin

I'm confused. We are missing more than just `stdin`: if we don't have atomics, we cannot call `.syncify()` from the worker, which means it's basically useless.

I think that as a first approximation we should be conservative: either allow the user code to run reliably, or we disallow it entirely:

- if we have the required headers and a browser which is new enough, we allow running in a worker
- if we don't, we display an error message with a good explanation

Reply:
Sounds good to me.

Reply:
Just a side question, for the purposes writing a release post and/or docs: will it be accurate to say:

> You can only use PyScript in a web worker if your server sets the "Cross-Origin-Embedder-Policy" header to `require-corp`, and the "Cross-Origin-Opener-Policy" header to `same-origin`. Otherwise, you will be limited to using PyScript in the main thread as before.

Reply:
I *think* it's accurate but I'll let Hood to answer because he knows more than me.
Note that my plan is the following:
- add a flag which determine whether to run in the main thread or in a worker (e.g. `execution-model = "mainthread"` or `execution-thread = "main" # or "worker")
- by default it will be "mainthread"
- you can choose "webworker", only if you meet the requirements
- in a future release, we can switch to "webworker" by default

Reply:
> side-question

Yes that's right for now.

Reply:
opened a follow-up issue: https://github.com/pyscript/pyscript/issues/1312